### DISCUSSION TITLE
IntroState and OverState triggers

### DISCUSSION THREAD

**Discussion by potsmugen** [ikemen-engine/Ikemen-GO#1172] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

The RoundState trigger is great, but it's also very limited when it comes to telling exactly in which part of which state you are in. 

### Describe the solution you'd like

We could use a couple more triggers:

IntroState:
0 - outside intros
1 - same as roundstate 0
2 - characters are doing their intros
3 - announcer is calling the round number
4 - fight screen

OverState, OutroState, WinState?
0 - outside win poses
1 - KO slowdown
2 - Slowdown is over and characters can still act
3 - characters have lost ctrl but aren't doing their win/lose poses yet
4 - characters doing their win/loseposes

These are just a rough draft, of course.

Alternatively there could be a new, more detailed, alternative to RoundState that'd include all of these phases.

### Additional context

This idea actually came from one of my pet peeves with RoundState, in that there is overlap between round states 1 and 2. When the lifebars show the "Fight!" text, the characters have no control but the game has already entered round state 2. I think that screen should still be round state 1.

**Comment by RealFoobs** [comment]
Seconding this as these triggers would make spawning explods and helpers at critical points so much easier and precise.

**Comment by K4thos** [comment, marked as answer]
#2284

**Comment by potsmugen** [comment]
In retrospect I think this implementation has some limitations:

- It is not explicit. i.e. it requires looking at the documentation to see what each number means;
- Only one state can be true at a time even though that's not the case in the engine;
- It can't be naturally expanded, i.e. if we need to reference more states in the future we'll have to find _some_ number to represent them, not necessarily ordered.

I had this other random idea for a way to bring it all together under the same trigger: FightScreenState (fight screen being the official term for lifebars). Usage examples:

FightScreenState(rounddisplay) -> True if showing round number
FightScreenState(koslowdown) -> True if applying KO slow motion
FightScreenState(windisplay) -> True if showing the winner names
Etc

(Names from memory. Would use same terms as fight.def

We could even throw the previously discussed sys.intro timer inside it for good measure since when you have a "super trigger" like this the bloat is much less of a problem.
